client/post-editor/editor-discussion/index.jsx

```diff
@@ -41,10 +41,13 @@ export default React.createClass( {
 			return this.props.post.discussion;
 		}
 
-		if ( this.props.site && this.props.isNew ) {
+		if ( this.props.site && this.props.isNew && this.props.post ) {
+			const { default_comment_status, default_ping_status } = this.props.site.options;
```

Comment:
Safer to use lodash get since we haven't checked for the existence of site options.
```jsx
import { get } from 'lodash';
//...
const defaultCommentStatus = get( site, 'options.default_comment_status', false );
const defaultPingStatus = get( site, 'options.default_ping_status', false );
```